processing/src/main/java/io/druid/segment/incremental/OakKeysComparator.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.incremental;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import io.druid.segment.DimensionIndexer;
+import com.oath.oak.OakComparator;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+public class OakKeysComparator implements OakComparator<IncrementalIndexRow>
```

Comment:
Lots of similar code down the road. Can the code be restructured into smaller methods, to be used in a modular way?